### PR TITLE
Windows support - 0.8.x

### LINKED ISSUE #53: Figure out and fix Windows compilation
So guys. Last year, I literally spent **2 months** trying to build Vega Strike on Windows 10! One of the most frustrating experiences of my developer life.

One thing I CAN tell you, is that Visual Studio 2019 is a no-go as a development environment. The required third-party dependencies just hadn't been updated to support VS2019. VS2017 is probably the latest version that will work.

We will also (probably) want to try clang as an alternate compiler on Windows. Perhaps llvm as well?

And finally, TBH, CMake may just not be up to the task any more. It is a venerable tool, and has served us well for years. But now, it might be time to look at other options? What do you guys think? I have a couple of suggestions for things we could try, if you guys want to hear them.

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
Closes #53 

Purpose:
- What is this pull request trying to do? Bring Windows support to the 0.8.0 release
- What release is this for? 0.8.0
- Is there a project or milestone we should apply this to? 0.8.x

### CODE REVIEW

File: engine/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "vega-strike",
+    "version-string": "0.9.0",
```
Comment (BenjamenMeyer) [inline engine/vcpkg.json:3]: This should be 0.8.0

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
One minor change needed.

File: engine/vcpkg.json (hunk 1 above)
Comment (stephengtuggy) [reply engine/vcpkg.json:3]: Oops. Thanks for catching that.

**Review by BenjamenMeyer — APPROVED** [review]
LGTM

**Pull request merged**